pkg/controller/ground.go

```diff
@@ -432,6 +451,16 @@ func (op *GroundControl) terminateKluster(kluster *v1.Kluster) error {
 		return err
 	}
 
+	if err := op.ensureFinalizerRemoved(kluster); err != nil {
+		return err
+	}
+
+	// There's a bug in the garbage-collector regarding CRDs in 1.7. It will not delete
+	// the CRD even though all Finalizers are gone. As a workaround, here we try to just
+	// delte the kluster again.
+	//
+	// This can be removed once the control-planes include garbage collector fixes
+	// for CDRs (1.8+)
```

Comment:
Can you maybe provide a link to the relevant PR fixing this bug in this comment?

Reply:
https://github.com/kubernetes/kubernetes/issues/50528

It's not quite clear where this is fixed. This explodes into various issues and fixes. They talk about that this is back-ported to 1.7 as well. It definitely is broken in our 1.7 Kubernikus control-planes.